### PR TITLE
jsc8 audit for Stream Workers using Rest API

### PR DESCRIPTION
**Pull request by koshy-thomas**
1. Removed the deprecated btoa functions instead used Buffer to convert to base64 string.
2. Correct the Output data variable for users to view results on console.
3. Did some very minor linting fixes and 2 lines of commented code to help user in case he uses a node version less than 17.5

### CODE REVIEW

File: docs/cep/tutorials/using_rest_api.md [hunk 1]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
-    class APIRequest {
-      _headers = {
-        Accept: "application/json",
-        "Content-Type": "application/json",
-      };
-
-      constructor(url) {
-        this._url = url;
-      }
+// If you are using Node. js version less than 17.5 then run "npm install node-fetch" and comment out the line below.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
// If you are using Node.js version less than 17.5, then run `npm install node-fetch` and comment out the line below.
```

File: docs/cep/tutorials/using_rest_api.md [hunk 2]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+const email = "nemo@nautilus.com";
+const password = "xxxxxx";
+const federationName = "api-gdn.paas.macrometa.io";
+const federationUrl = `https://${federationName}`;
+
+const isGlobal = false;
+const stream = "tutorialAppInputStream";
+const streamAppName = "stream_app_tutorial";
+const streamAppDef = `@App:name('stream_app_tutorial')
+  @App:description('This application demonstrates how to use user defined functions in the stream app')
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
  @App:description('This application demonstrates how to use user-defined functions in the stream app.')
```

File: docs/cep/tutorials/using_rest_api.md [hunk 3]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+  @info(name='Query')
+  INSERT INTO tutorialAppOutputTable
+  SELECT concatFn(roomNo,'-',deviceID) as id, temperature
+  FROM tutorialAppInputStream;`;
+
+const run = async function () {
+  try {
+    const connection = new APIRequest(federationUrl);
+
+    /* -------------------- Login (nemo@nautilus.com/xxxxxx) -------------------- */
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
    /* -------------------- Log in (nemo@nautilus.com/xxxxxx) -------------------- */
```

File: docs/cep/tutorials/using_rest_api.md [hunk 4]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+    }
 
-      _handleResponse(response, resolve, reject) {
-        if (response.ok) {
-          resolve(response.json());
-        } else {
-          reject(response);
-        }
-      }
+    /* --------------------------- Activate StreamApp --------------------------- */
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]: I am confused. In some places, you use StreamApp, in others "stream app." Why the two spellings?

File: docs/cep/tutorials/using_rest_api.md [hunk 5]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
       }
+    );
+
+    console.log("Activating stream app...", streamAppName);
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+
+    console.log("Stream app activated successfully");
+
+    /* ------------------ Publish messages to Sample StreamApp ------------------ */
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
    /* ------------------ Publish messages to sample StreamApp ------------------ */
    
```

File: docs/cep/tutorials/using_rest_api.md [hunk 6]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+    const region = isGlobal ? "c8global" : "c8local";
+    const streamName = `${region}s.${stream}`;
+
+    // Fetching local URL in case the stream is local (which is defined in the streamapp)
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]: And here is a third spelling of "streamapp"

File: docs/cep/tutorials/using_rest_api.md [hunk 7]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+      ? federationName
+      : `api-${dcUrl}`;
+
+    const otpProducer = await connection.req(`/apid/otp`, {
+      method: "POST"
+    });
+
+    const producerUrl = `wss://${url}/_ws/ws/v2/producer/persistent/${tenant}/${region}._system/${streamName}?otp=${otpProducer.otp}`;
+
+    /* -------------------------- Initalizing Producer -------------------------- */
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
    /* -------------------------- Initalizing producer -------------------------- */
```

File: docs/cep/tutorials/using_rest_api.md [hunk 8]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+      })
+    );
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+
+    producer.close();
+
+    /* ----------------------------- Verify results ----------------------------- */
+
+    const selectQuery = "FOR doc IN tutorialAppOutputTable return doc";
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]: Is this a C8QL? If so, shouldn't RETURN be all caps?

File: docs/cep/tutorials/using_rest_api.md [hunk 9]
```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
...
+        query: selectQuery,
+        bindVars: {}
+      },
+      method: "POST"
+    });
+
+    console.log("Input sent --->", inputData);
+    console.log("Output received --->", result.result);
+
+    /* ---------------------------- Delete StreamApp ---------------------------- */
```
Comment (diana-macrometa) [inline docs/cep/tutorials/using_rest_api.md]:
```suggestion
    /* ---------------------------- Delete StreamApp ---------------------------- */
    
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

File: docs/cep/tutorials/using_rest_api.md (hunk 8 above)
Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: c8ql supports caps and lowercases both actually

Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: anyways I will make it caps for consistency

File: docs/cep/tutorials/using_rest_api.md (hunk 4 above)
Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: It was already there as part of the code I didn't change these actually

File: docs/cep/tutorials/using_rest_api.md (hunk 6 above)
Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: This one I missed yeah.. I will change it

Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: Thanks for the reviews

File: docs/cep/tutorials/using_rest_api.md (hunk 4 above)
Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: done

File: docs/cep/tutorials/using_rest_api.md (hunk 8 above)
Comment (koshy-thomas) [reply docs/cep/tutorials/using_rest_api.md]: done

**Comment by koshy-thomas** [comment]
@diana-macrometa Thanks for the reviews! :)

**Review by diana-macrometa — APPROVED** [review]
LGTM

**Pull request merged**